### PR TITLE
:seedling: Unitetst for GitLab workflow support and tests

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add a test file for Gitlab workflows
- Add a new file `clients/gitlabrepo/testdata/valid-project-jobs`
- Delete the `workflow_e2e_test.go` file and remove the code related to the GitLab webhooks test

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: clients/gitlabrepo/workflows_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			handler := &workflowsHandler{
+				glClient: client,
+			}
+
+			repoURL := repoURL{
+				owner:     "ossf-tests",
+				commitSHA: clients.HeadSHA,
+			}
+			handler.init(&repoURL)
+			got, err := handler.listSuccessfulWorkflowRuns("artifacts.zip")
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply clients/gitlabrepo/workflows_test.go:75]: A friendly ping

[the remark this answers is not in the document]
Comment (raghavkaul) [reply clients/gitlabrepo/workflows_test.go:75]: GitLab defines all CI/CD workflows in a single yml file (`[.gitlab-ci.yml`](https://docs.gitlab.com/ee/ci/yaml/gitlab_ci_yaml.html)) while GitHub allows any number of workflows to be defined in the `.github` directory. `gitlabrepo.ListSuccessfulWorkflowRuns` needs to conform to the `Client` interface, but this helper method shouldn't take any parameter, really.

The only usage of `gitlabrepo.ListSuccessfulWorkflowRuns` is in `raw.BinaryArtifacts` when looking for Gradle wrapper validation workflows. We don't have a equivalent for GitLab (though one proposal is [in draft](https://github.com/gradle/wrapper-validation-action/pull/50/files)). Unless there's an alternative, all GitLab repos with gradle binaries are going to have lower Binary-Artifacts scores without an easy remediation.

In conclusion, this helper function is acting as dead code right now.
